src/org/zaproxy/zap/extension/ascanrulesBeta/HPP.java

```diff
@@ -154,7 +157,12 @@ public void scan() {
 					log.debug("Injecting payload...");
 					HttpMessage newMsg = getNewMsg();
 					newMsg.setGetParams(tags);
-					sendAndReceive(newMsg);
+					try {
+						sendAndReceive(newMsg);
+					} catch (IllegalStateException|UnknownHostException ex) {
+						log.info("Caught " + ex.getClass().getName() + " when accessing: " + newMsg.getRequestHeader().getURI().toString() + 
+								"\n The target may have replied with a poorly formed redirect due to our input.");
```

Comment:
Added a return to break out of further work. Here's an example of it tested against wavsep:

> 144526 [ZAP-ActiveScanner-1] INFO org.zaproxy.zap.extension.ascanrulesBeta.HPP  - Caught java.net.UnknownHostException when accessing: http://localhost:8081/active/Unvalidated-Redirect/Redirect-Detection-Evaluation-POST-302Redirect/Case07-Redirect-RedirectMethod-FilenameContext-Unrestricted-HttpURL-DefaultEmptyInput-PartialPathReq-Read.jsp?target=%2526zap%253Dzaproxy
 The target may have replied with a poorly formed redirect due to our input.

My interpretation of the code and result is that the response is the result of HPP's "attack" ... Though the redirect being invalid is due to wavsep's handling not ZAP's. Does that answer your question?